src/Simplify.elm

```diff
@@ -3944,19 +3966,95 @@ listHeadChecks checkInfo =
-            case singletonCall.argsAfterFirst of
+listTailExistsError : { message : String, details : List String }
+listTailExistsError =
+    { message = "Using List.tail on a list with some elements will result in Just the elements after the first"
+    , details = [ "You can replace this call by Just the list elements after the first." ]
+    }
+
+
+listEmptyTailExistsError : { message : String, details : List String }
+listEmptyTailExistsError =
+    { message = "Using List.tail on a list with a single element will result in Just the empty list"
+    , details = [ "You can replace this call by Just the empty list." ]
+    }
+
+
+listTailChecks : CheckInfo -> List (Error {})
+listTailChecks checkInfo =
+    let
+        listArg : Node Expression
+        listArg =
+            AstHelpers.removeParens checkInfo.firstArg
+
+        listArgRange : Range
+        listArgRange =
+            Node.range listArg
+    in
+    case Node.value listArg of
+        Expression.ListExpr listLiteral ->
+            case listLiteral of
                 [] ->
-                    Just { element = singletonCall.firstArg }
+                    [ Rule.errorWithFix
+                        { message = "Using List.tail on an empty list will result in Nothing"
+                        , details = [ "You can replace this call by Nothing." ]
+                        }
+                        checkInfo.fnRange
+                        [ Fix.replaceRangeBy checkInfo.parentRange "Nothing" ]
+                    ]
 
-                _ :: _ ->
-                    Nothing
+                _ :: [] ->
+                    [ Rule.errorWithFix
+                        listEmptyTailExistsError
+                        checkInfo.fnRange
+                        [ Fix.replaceRangeBy listArgRange listCollection.emptyAsString
+                        , Fix.replaceRangeBy checkInfo.fnRange "Just"
+                        ]
+                    ]
 
-        Nothing ->
-            Nothing
+                _ :: (Node tailFirstRange _) :: tailAfterFirst ->
+                    let
+                        tailLastRange : Range
+                        tailLastRange =
+                            Maybe.withDefault tailFirstRange (lastElementRange tailAfterFirst)
+
+                        tailRange : Range
+                        tailRange =
+                            { start = tailFirstRange.start, end = tailLastRange.end }
+                    in
+                    [ Rule.errorWithFix
+                        listTailExistsError
+                        checkInfo.fnRange
+                        (keepOnlyFix { parentRange = listArgRange, keep = tailRange }
+                            ++ [ Fix.insertAt tailFirstRange.start "[ "
+                               , Fix.insertAt tailLastRange.end " ]"
+                               , Fix.replaceRangeBy checkInfo.fnRange "Just"
+                               ]
+                        )
+                    ]
```

Comment:
I think a simpler fix would be to remove the first element. So if you have `[ a, b, c ]`, remove `a, `.
```suggestion
                Node headRange _ :: (Node tailFirstRange _) :: _ ->
                    [ Rule.errorWithFix
                        listTailExistsError
                        checkInfo.fnRange
                        [ Fix.removeRange { start = headRange.start, end = tailFirstRange.start }
                        , Fix.replaceRangeBy checkInfo.fnRange "Just"
                        ]
                    ]
```
(I did not test the suggestion)

Reply:
That's so clean